Content.Server/Clothing/Components/RestrictedClothingComponent.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Content.Server.Speech.Components;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Clothing.Components;
+
+/// <summary>
+///     Does bad things when someone tries to wear something they shouldn't
+/// </summary>
+[RegisterComponent]
+public sealed class RestrictedClothingComponent : Component
+{
+    /// <summary>
+    ///     List of permissions required to be allowed to wear the item
+    /// </summary>
+    [DataField("permissions")] [ViewVariables(VVAccess.ReadWrite)]
+    public HashSet<string> Permissions = new();
```

Comment:
cant you make this a hashset of admin flags and then you dont have to parse it in the system?

Reply:
AdminFlags is a flag so you can just use public AdminFlags and mask it.